File: scratch/fix_registry.py
import re

path = "src/components/exercise/courseExerciseFinalBatchRegistry.ts"
with open(path, "r") as f:
    content = f.read()

# Fix the syntax error
content = content.replace("}  : Partial<\n  Record<CourseExerciseCategoryEnum, CourseExerciseCategoryConfig>\n>;", "};")
content = content.replace(
    "export const FINAL_BATCH_CATEGORY_CONFIGS = {",
    "export const FINAL_BATCH_CATEGORY_CONFIGS: Partial<Record<CourseExerciseCategoryEnum, CourseExerciseCategoryConfig>> = {"
)

with open(path, "w") as f:
    f.write(content)

File: scratch/fix_registry.js
const fs = require('fs');
const file = 'src/components/exercise/courseExerciseFinalBatchRegistry.ts';
let code = fs.readFileSync(file, 'utf8');

// Fix the corrupted `getCheckpointTransition`
code = code.replace(/state_switch: StateSwitchConfig,\n};/, '};');

// Add to the actual FINAL_BATCH_CATEGORY_CONFIGS
code = code.replace(/\[CourseExerciseCategoryEnum\.SectionMilestone\]: SectionMilestoneConfig,\n};/, '[CourseExerciseCategoryEnum.SectionMilestone]: SectionMilestoneConfig,\n  [CourseExerciseCategoryEnum.StateSwitch]: StateSwitchConfig,\n};');

fs.writeFileSync(file, code);
